types/transaction_helpers.go

```diff
@@ -0,0 +1,110 @@
+// +build testing
+
+package types
+
+import (
+	"errors"
+)
+
+// TransactionGraph will return a set of valid transactions that all spend
+// outputs according to the input graph. Each [source, dest] pair defines an
+// edge of the graph. The graph must be fully connected and the granparent of
+// the graph must be the sourceOutput. '0' refers to an edge from the source
+// output. Each edge also specifies a value for the output, and an amount of
+// fees. If the fees are zero, no fees will be added for that edge. 'sources'
+// must be sorted.
+//
+// Example of acceptable input:
+//
+// sourceOutput: // a valid siacoin output spending to UnlockConditions{}.UnlockHash()
+//
+// Sources: [0, 0, 1, 2, 3, 3, 3, 4]
+// Dests:   [1, 2, 3, 3, 4, 4, 5, 6]
+//
+// Resulting Graph:
+//
+//    o
+//   / \
+//  o   o
+//   \ /
+//    o
+//   /|\
+//   \| \
+//    o  x // 'x' transactions are symbolic, not actually created
+//    |
+//    x
+//
+func TransactionGraph(sourceOutput SiacoinOutputID, sources []int, dests []int, outputValues []Currency, fees []Currency) ([]Transaction, error) {
```

Comment:
why not create a new type instead of four separate slices?
```go
type TransactionGraphEdge struct {
    Input      *TransactionGraphEdge
    Output     *TransactionGraphEdge
    Value, Fee types.Currency
}
```
Then you wire up the edges yourself, at the callsite, and the `TransactionGraph` function just converts them into a real transaction.
```go
edges := make([]types.TransactionGraphEdge, 3)
edges[0].Value = types.SiacoinPrecision
edges[0].Output = &edges[1]
edges[1].Value = types.SiacoinPrecision
edges[1].Output = &edges[2]
types.TransactionGraph(sourceOutput, edges)
```
(definitely some kinks in this that I haven't fully thought out, but you get the idea)